core/src/main/resources/bootstrap/tuningConfigs.yaml

```diff
@@ -0,0 +1,275 @@
+# - 'qualification': Tool-specific overrides for the Qualification Tool
+# - 'profiling': Tool-specific overrides for the Profiling Tool
+#
+# Each configuration entry contains:
+# - name: The configuration parameter name/key
+# - description: Human-readable description of the configuration parameter
+# - default: The default value to use
+# - min: The minimum allowed value (optional)
+# - max: The maximum allowed value (optional)
+# - usedBy: Comma-separated list of Spark configurations this constant affects
+#
+# Usage: Users can override these default values by specifying custom tuning configurations via:
+# - `--tuning-configs /path/to/custom/file.YAML` argument when running the Tools JAR
+# - `--tuning_configs /path/to/custom/file.YAML` argument when running the Tools CLI
+
+default:
+  - name: CONC_GPU_TASKS
+    description: >-
+      Maximum number of concurrent GPU tasks that can run simultaneously on a single GPU
+    max: 4
+    usedBy: spark.rapids.sql.concurrentGpuTasks
+
+  - name: CORES_PER_EXECUTOR
+    description: >-
+      Default number of CPU cores per executor
+    default: 16
+    usedBy: spark.executor.cores
+
+  - name: TASK_GPU_RESOURCE_AMT
+    description: >-
+      Amount of GPU resource each task requires (fraction of GPU)
+    default: 0.001
+    usedBy: spark.task.resource.gpu.amount
+
+  - name: HEAP_OVERHEAD_FRACTION
+    description: >-
+      Fraction of executor memory reserved for JVM overhead
+    default: 0.1
+    usedBy: spark.executor.memoryOverhead
+
+  - name: JVM_GCTIME_FRACTION
+    description: >-
+      Maximum acceptable fraction of time spent in JVM garbage collection
+    max: 0.3
+    usedBy: N/A
+
+  # Memory configuration parameters
+  - name: HEAP_PER_CORE
+    description: >-
+      Amount of heap memory allocated per CPU core
+    default: 2g
+    min: 750m
+    usedBy: spark.executor.memory
+
+  - name: PINNED_MEMORY
+    description: >-
+      Amount of pinned memory for GPU operations
+    default: 2g
+    max: 4g
+    usedBy: spark.rapids.memory.pinnedPool.size
+
+  - name: SPILL_MEMORY
+    description: >-
+      Amount of memory reserved for spilling operations.
+    default: 2g
+    usedBy: spark.rapids.memory.spillPool.size
+
+  - name: GPU_MEM_PER_TASK
+    description: >-
+      Amount of GPU memory to use per concurrent task (on a single GPU).
+      This estimate is based on T4s. This will be used to calculate the number of
+      concurrent GPU tasks.
+    default: 7500m
+    usedBy: spark.rapids.sql.concurrentGpuTasks
+
+  - name: INPUT_BYTES_RANGE
+    description: >-
+      Target range for input bytes per partition. This will be used to
+      calculate maxPartitionBytes.
+    min: 128m
+    max: 256m
+    usedBy: spark.sql.files.maxPartitionBytes
+
+  - name: MAX_PARTITION_BYTES
+    description: >-
+      Maximum partition size for file reads
+    default: 512m
+    max: 4g
+    usedBy: spark.sql.files.maxPartitionBytes
```

Comment:
I'm confused by these 2 configs. Do they conflict?
Above range is from 128m to 256m, but below says default 512m, max is 4g.

Reply:
Config `INPUT_BYTES_RANGE` was used to define a threshold range for the task input size.

Renamed the config `INPUT_BYTES_RANGE` to `TASK_INPUT_SIZE_THRESHOLD` for more clarity and improved the description to state its usage.

```yaml
  - name: TASK_INPUT_SIZE_THRESHOLD
    description: >-
      Threshold range for task input size used to determine when to adjust maxPartitionBytes.
      If actual task input size is below min threshold, maxPartitionBytes is increased.
      If actual task input size is above max threshold, maxPartitionBytes is decreased.
    min: 128m
    max: 256m
    usedBy: spark.sql.files.maxPartitionBytes
```

Reply:
What's the difference between task input size vs maxPartitionBytes if the task here is in the first scan stage? If the task is not the scan stage, I think that'll be controlled by shuffle.partitions, no related to maxPartitionBytes.

Reply:
From my understanding: task input size is the actual data read by the scan task at runtime, which can be different from maxPartitionBytes due to predicate pushdown, column pruning, or decompression. For example:
- If due to predicate pushdown or column pruning, the actual input size is less than 128m, we can increase `maxPartitionBytes` because we know some of the data is going to be pruned.
- If due to decompression, the actual input size is more than 256m, we should decrease `maxPartitionBytes` so that Spark creates smaller file size partitions at beginning.

Reply:
Discussed with @GaryShen2008, will continue this discussion offline.